### PR TITLE
Gh 219 schema group inheritance

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

File: core/store/src/test/java/uk/gov/gchq/gaffer/store/schema/SchemaTest.java [hunk 1]
```diff
@@ -58,19 +57,16 @@
 import static org.junit.Assert.fail;
 import static org.mockito.Mockito.mock;
 
+
 public class SchemaTest {
```
Comment (t616178) [inline core/store/src/test/java/uk/gov/gchq/gaffer/store/schema/SchemaTest.java]: Are there any tests for serialising/deserialising JSON schemas containing parent types? Or any examples of this for the wiki?

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/Schema.java [hunk 2]
```diff
@@ -302,109 +200,230 @@ public String toString() {
...
+        }
+    }
+
+    @JsonPOJOBuilder(buildMethodName = "build", withPrefix = "")
+    public static final class Builder extends BaseBuilder<Builder> {
+        public Builder() {
+        }
+
+        public Builder(final Schema schema) {
+            this();
```
Comment (p013570) [inline core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/Schema.java]: I think it is useful - it is just a shortcut for new Builder().merge(schema)
Allows users to start from a existing schema and make change.

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java [hunk 3]
```diff
@@ -92,57 +96,92 @@ public View clone() {
...
+    @JsonPOJOBuilder(buildMethodName = "build", withPrefix = "")
+    public static final class Builder extends BaseBuilder<Builder> {
+        public Builder() {
         }
 
-        @Override
-        public View build() {
-            return (View) super.build();
+        public Builder(final View view) {
+            this();
```
Comment (p013570) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java]: I think it is useful - it is just a shortcut for new Builder().merge(view)
Allows users to start from a existing view and make changes.

File: core/store/src/test/java/uk/gov/gchq/gaffer/store/schema/SchemaTest.java (hunk 1 above)
Comment (p013570) [inline core/store/src/test/java/uk/gov/gchq/gaffer/store/schema/SchemaTest.java]: Thanks - I'll update the getting started guide with an example.

**Comment by p013570** [comment]
Closing pull request until I have made the changes needed.

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java (hunk 3 above)
Comment (t616178) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java]: Sorry, I mean the call onto this() (the no-arg constructor) which doesn't contain a body.

**Pull request merged**